pallets/creditcoin/src/types/platform.rs

```diff
@@ -20,25 +22,63 @@ impl From<u64> for EvmChainId {
 }
 
 impl EvmChainId {
-	pub fn new(value: u64) -> Self {
+	pub const fn new(value: u64) -> Self {
 		EvmChainId(value)
 	}
 }
 
 #[derive(
 	Clone, RuntimeDebug, PartialEq, Eq, PartialOrd, Encode, Decode, TypeInfo, MaxEncodedLen,
 )]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
 pub struct EvmInfo {
 	pub chain_id: EvmChainId,
 }
 
 #[derive(
 	Clone, RuntimeDebug, PartialEq, Eq, PartialOrd, Encode, Decode, TypeInfo, MaxEncodedLen,
 )]
-pub enum NewBlockchain {
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub enum Blockchain {
 	Evm(EvmInfo),
 }
 
+impl Blockchain {
+	pub const fn evm(chain_id: EvmChainId) -> Blockchain {
+		Blockchain::Evm(EvmInfo { chain_id })
+	}
+	pub const ETHEREUM: Blockchain = Blockchain::evm(EvmChainId::new(1));
+	pub const RINKEBY: Blockchain = Blockchain::evm(EvmChainId::new(4));
+	pub const LUNIVERSE_TESTNET: Blockchain = Blockchain::evm(EvmChainId::new(1635501961136826136));
+	pub const LUNIVERSE: Blockchain = Blockchain::evm(EvmChainId::new(3158073271666164067));
+
+	pub fn as_bytes(&self) -> &[u8] {
+		match self {
+			&Blockchain::ETHEREUM => b"ethereum",
+			&Blockchain::RINKEBY => b"rinkeby",
+			&(Blockchain::LUNIVERSE_TESTNET | Blockchain::LUNIVERSE) => b"luniverse",
+			_ => todo!(),
+		}
+	}
+
+	pub fn supports(&self, kind: &LegacyTransferKind) -> bool {
+		match (self, kind) {
+			(
+				&Blockchain::ETHEREUM
+				| &Blockchain::RINKEBY
+				| &Blockchain::LUNIVERSE
+				| &Blockchain::LUNIVERSE_TESTNET,
+				LegacyTransferKind::Erc20(_)
+				| LegacyTransferKind::Ethless(_)
+				| LegacyTransferKind::Native,
```

Comment:
Do we really support all of these transfers atm, @jfrank-summit?

Reply:
This is more about whether the chain supports the transfer kind, not whether we do (though I agree it's weird, I just kept this to retain the old behavior. It will only matter for Aella anyway because all new transfers will use the new TransferKind).

We still return an error for legacy transfer kinds we don't actually support in the register_transfer extrinsics, it's just a separate check from this function.